### PR TITLE
Pass the network context to the s2i builder

### PR DESCRIPTION
**Pull request by pedro-r-marques**
Make the builder pod pass the network context to the source-to-image builder pod.

### CODE REVIEW

**Comment by pedro-r-marques** [comment]
@rajatchopra @bparees @mfojtik This PR uses the DockerNetworkMode setting introduced in https://github.com/openshift/source-to-image/pull/311 in order to execute the s2i builder in the context of the infrastructure container of the builder pod.

**Comment by bparees** [comment]
for the source to image deps update, please follow our dep bumping commit syntax:
`bump(github.com/openshift/source-to-image) <commitid>`

see commit 9f28d8a890573e36997dbaaa58c2f5f0b65308d5 for example (the last time we updated this dep)

**Comment by bparees** [comment]
otherwise the s2i update commit lgtm.  i'm abstaining on the other part of this PR.

**Comment by bparees** [comment]
actually I take that back.  please don't remove the _test files from the commit, everything should be bumped.

**Comment by smarterclayton** [comment]
LGTM [merge]

File: pkg/build/builder/util.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	stiapi "github.com/openshift/source-to-image/pkg/api"
+)
+
+// readNetClsCGroup parses /proc/self/cgroup in order to determine the container id that can be used
+// the network namespace that this process is running on.
+func readNetClsCGroup(reader io.Reader) string {
+	cgroups := make(map[string]string)
+
+	re := regexp.MustCompile(`\d+:([a-z_,]+):/.*/(docker-|)([a-z0-9]+).*`)
```
Comment (smarterclayton) [inline pkg/build/builder/util.go]: Actually one minor tweak, move this to a large level

**Comment by smarterclayton** [comment]
Package level, that is.

**Pull request merged**